### PR TITLE
Add AclRegister interfaces and ClusterAccessor integration

### LINKED ISSUE #2152: Add AclRegister to CreateCluster API endpoint
**Is your feature request related to a problem? Please describe.**
We need an AclRegister class so that ACL for clusters are automatically created when clusters are created. 

**Describe the solution you'd like**
Create an AclRegister interface that can be implemented and called during CreateCluster. 

**Additional context**

### PR DESCRIPTION
**Pull request by NealSun96**
### Issues

- [x] My PR addresses the following Helix issues and references them in the PR description:

Fixes #2152 

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

This PR introduces the AclRegister interface, an interface that can be implemented with developers' choices of mechanisms on how ACL is created and registered. The implementation is then registered to the rest server, and it's called when clusters are created. 

### Tests

- [x] The following tests are written for this issue:

```
TestAclRegister
```

- The following is the result of the "mvn test" command on the appropriate module:

(If CI test fails due to known issue, please specify the issue and test PR locally. Then copy & paste the result of "mvn test" to here.)

### Changes that Break Backward Compatibility (Optional)

- My PR contains changes that break backward compatibility or previous assumptions for certain methods or API. They include:

Rerun failed test in CI:
```
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 62.693 s - in org.apache.helix.integration.task.TestMaxNumberOfAttemptsMasterSwitch
[INFO]
[INFO] Results:
[INFO]
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
```

(Consider including all behavior changes for public methods or API. Also include these changes in merge description so that other developers are aware of these changes. This allows them to make relevant code changes in feature branches accounting for the new method/API behavior.)

### Documentation (Optional)

- In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

File: helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/ClusterAccessor.java [hunk 1]
```diff
@@ -187,6 +189,13 @@ public Response createCluster(@PathParam("clusterId") String clusterId,
       LOG.error("Failed to create cluster {}. Exception: {}.", clusterId, ex);
       return serverError(ex);
     }
+
+    try {
```
Comment (rabashizade) [inline helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/ClusterAccessor.java:193]: If we perform this before the cluster creation, then if ACL fails the cluster isn't created either. I'm not sure which one is the better option though, a cluster without ACL or an ACL without a cluster.

**Review by rabashizade — CHANGES_REQUESTED** [review]

File: helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/ClusterAccessor.java (hunk 1 above)
Comment (NealSun96) [reply helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/ClusterAccessor.java:193]: Synced offline: we will allow only allow a cluster to be created if the ACL exists, so we are changing the order. 

When creating ACL, we will keep it idempotent, but only for the correct user.

File: helix-rest/src/main/java/org/apache/helix/rest/acl/AclRegister.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package org.apache.helix.rest.acl;
```
Comment (desaikomal) [inline helix-rest/src/main/java/org/apache/helix/rest/acl/AclRegister.java:1]: nit: do you need to add the standard Apache banner?

File: helix-rest/src/main/java/org/apache/helix/rest/acl/AclRegister.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+
+import javax.servlet.http.HttpServletRequest;
+
+
+public interface AclRegister {
```
Comment (desaikomal) [inline helix-rest/src/main/java/org/apache/helix/rest/acl/AclRegister.java:6]: nit: Please add some comments about purpose of this interface.

File: helix-rest/src/test/java/org/apache/helix/rest/server/TestAclRegister.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+            new NoopAuthValidator(), new NoopAuthValidator(), mockAclRegister);
+    server.start();
+
+    HttpUriRequest request =
+        buildRequest("/clusters/testCluster", HttpConstants.RestVerbs.PUT, "");
+    sendRequestAndValidate(request, Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+    request =
+        buildRequest("/clusters/testCluster", HttpConstants.RestVerbs.GET, "");
+    sendRequestAndValidate(request, Response.Status.NOT_FOUND.getStatusCode());
+
```
Comment (desaikomal) [inline helix-rest/src/test/java/org/apache/helix/rest/server/TestAclRegister.java:93]: how about proper successful test case?

File: helix-rest/src/main/java/org/apache/helix/rest/acl/AclRegister.java [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+  void createACL(HttpServletRequest request);
```
[the remark this answers is not in the document]
Comment (NealSun96) [reply helix-rest/src/main/java/org/apache/helix/rest/acl/AclRegister.java:7]: Those functionalities can be supported by other interfaces. For this PR, I'm adding the minimal required functionality which is to register an ACL entry.

File: helix-rest/src/test/java/org/apache/helix/rest/server/TestAclRegister.java (hunk 4 above)
Comment (NealSun96) [reply helix-rest/src/test/java/org/apache/helix/rest/server/TestAclRegister.java:93]: The first test case covered that.

**Review by desaikomal — APPROVED** [review]
LGTM..

**Review by rabashizade — APPROVED** [review]
Thank you for addressing the comments.

**Pull request merged**